src/PrestaShopBundle/Form/Admin/Catalog/Category/AbstractCategoryType.php

```diff
@@ -67,51 +70,87 @@ abstract class AbstractCategoryType extends TranslatorAwareType
      */
     private $configuration;
 
+    /**
+     * @var Router
+     */
+    private $router;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
     /**
      * @param TranslatorInterface $translator
      * @param array $locales
      * @param array $customerGroupChoices
      * @param FeatureInterface $multistoreFeature
      * @param ConfigurationInterface $configuration
+     * @param Router $router
+     * @param CommandBusInterface $queryBus
      */
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         array $customerGroupChoices,
         FeatureInterface $multistoreFeature,
-        ConfigurationInterface $configuration
+        ConfigurationInterface $configuration,
+        Router $router,
+        CommandBusInterface $queryBus
     ) {
         parent::__construct($translator, $locales);
 
         $this->customerGroupChoices = $customerGroupChoices;
         $this->multistoreFeature = $multistoreFeature;
         $this->configuration = $configuration;
+        $this->router = $router;
+        $this->queryBus = $queryBus;
     }
 
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $disableMenuThumbnailsUpload = false;
+        if (isset($options['id_category'])) {
+            $disableMenuThumbnailsUpload = $options['disable_menu_thumbnails_upload'];
+        }
+        $genericCharactersHint = $this->trans('Invalid characters:', 'Admin.Global') . ' <>;=#{}';
+
         $builder
             ->add('name', TranslatableType::class, [
+                'label' => $this->trans('Name', 'Admin.Global'),
+                'help' => $genericCharactersHint,
                 'type' => TextType::class,
                 'constraints' => [
                     new DefaultLanguage(),
                 ],
                 'options' => [
+                    'attr' => [
+                        'maxlength' => self::MAX_TITLE_LENGTH,
+                    ],
                     'constraints' => [
                         new Regex([
                             'pattern' => '/^[^<>;=#{}]*$/u',
                             'message' => $this->trans('%s is invalid.', 'Admin.Notifications.Error'),
                         ]),
+                        new Length([
+                            'max' => self::MAX_TITLE_LENGTH,
+                            'maxMessage' => $this->trans(
+                                'This field cannot be longer than %limit% characters.',
+                                'Admin.Notifications.Error',
+                                [
+                                    '%limit%' => SeoSettings::MAX_TITLE_LENGTH,
```

Comment:
```suggestion
                                    '%limit%' => static::MAX_TITLE_LENGTH,
```
Otherwise there is something wrong with the `MAX_TITLE_LENGTH` usage